modules/message-lane/src/instant_payments.rs

```diff
@@ -55,34 +65,252 @@ where
 		}
 	}
 
-	fn pay_relayer_reward(
-		_confirmation_relayer: &AccountId,
-		relayer: &AccountId,
-		reward: &Currency::Balance,
+	fn pay_relayers_rewards(
+		confirmation_relayer: &AccountId,
+		relayers_rewards: RelayersRewards<AccountId, Currency::Balance>,
 		relayer_fund_account: &AccountId,
 	) {
-		let pay_result = Currency::transfer(
-			&relayer_fund_account,
-			relayer,
-			*reward,
-			ExistenceRequirement::AllowDeath,
+		pay_relayers_rewards(
```

Comment:
This `InstantCurrency` trait is just for tests simplicity. All currency methods are static, so without `&mut self` you have to use thread locals ot something like that, which I prefer to avoid.